pecos/core/ann/hnsw.hpp

```diff
@@ -238,6 +243,143 @@ namespace ann {
         }
     };
 
+    struct GraphProductQuantizer4Bits : GraphBase {
+        typedef FeatVecDenseL2Simd<float> feat_vec_t;
+
+        ProductQuantizer4Bits quantizer;
+        index_type num_node;
+        index_type code_dimension;
```

Comment:
is this equivalent to `local_dimension` as defined `quantizer struct`?

Reply:
local_dimension is the number of dimensions of subspace, but code_dimension is the code length encoded in the graph for a single data point. code_dimension is the padded num_local_codebooks, and as explained above I think it's better not to rename.